go/vt/vtgate/tabletgateway.go

```diff
@@ -300,6 +300,9 @@ func (gw *TabletGateway) withRetry(ctx context.Context, target *querypb.Target,
 		for _, t := range tablets {
 			if _, ok := invalidTablets[topoproto.TabletAliasString(t.Tablet.Alias)]; !ok {
 				th = t
+				if *routeReplicaToRdonly && (target.TabletType == topodatapb.TabletType_REPLICA || target.TabletType == topodatapb.TabletType_RDONLY) {
```

Comment:
I thought target.TabletType can be either TabletType_PRIMARY  or TabletType_REPLICA, and never be TabletType_RDONLY as here target is passed-in on line 221?

Reply:
if we just update line 327 with `inner(ctx, th.Target, th.Conn)`, would it work?